shared/js/moz_intl.js

```diff
@@ -352,6 +373,34 @@ global.mozIntl = {
         },
       };
     },
+
+    getFormattedUnit: function(type, style, v) {
+      if (!unitFormatData.hasOwnProperty(type)) {
+        throw new RangeError(`invalid type ${type}`);
```

Comment:
That's how https://tc39.github.io/ecma402/ deals with options that are not in range. You can test for yourself:

```
(42).toLocaleString('en-US', {style: 'foo'});
RangeError: invalid value foo for option style
```